packToken.h

```diff
@@ -76,6 +76,13 @@ class packToken {
  public:
   // Used to recover the original pointer.
   // The intance whose pointer was removed must be an rvalue.
+  TokenBase* release() & {
+    TokenBase* b = base;
+    // Setting base to 0 leaves the class in an invalid state,
+    // except for destruction.
+    base = 0;
+    return b;
+  }
```

Comment:
If I remember correctly this function signature `TokenBase* release() &` was omitted on purpose in order to make it harder to produce an invalid packToken by accident.

The only way to call it would be to explicitly call `std::move(myPackToken).release()`. This would prevent anyone from using it recklessly.

I don't mind removing this safe-guard, but it would be best if we had a sample code to test if this change really is necessary.

Reply:
To better illustrate the risk I am referring to, take a look at this sample code:

```C++
#include "shunting-yard.h"

int main() {
  packToken p = 10;
  p.release();
  p.asInt(); // Seg. fault
}
```